### PR TITLE
Add Streamlit Chat Migration Guide

### PR DESCRIPTION
**Pull request by MarcSkovMadsen**
Builds on top of #5633.

This PR adds a Streamlit Chat Migration Guide. It will

- Make the Streamlit Migration Guide more complete
- Benchmark our chat components against Streamlits
- Provide the `StreamlitConverter` app example some *context* about conversational chat apps. See https://github.com/holoviz-topics/panel-chat-examples/pull/61.

Preview the guide [here](https://github.com/holoviz/panel/blob/enhancement/migrate-streamlit-chat/doc/how_to/streamlit_migration/chat.md)

FYI. @ahuang11 

## Questions

- [x] Should we rename `ChatEntry` to `ChatMessage`. Both Streamlit and LangChain use the term *message*. For example `st.chat_message`. The `ChatEntry` name could also mistakingly be interpreted as a `chat_input` like component?
- [ ] Should we provide a `ChatInput` component similar to Streamlits `chat_input`?
- [ ] Should we provide a `Status` component similar to Streamlits `status`?

## Todo:

- [ ] Merge with main and rename ChatEntry to ChatMessage. Rename ChatMessage.value to .object and ChatFeed.value to .objects.
- [ ] Philipp should upload the assets and replace the links.

### CODE REVIEW

File: doc/_static/images/streamlit_chat_input.png
Comment (MarcSkovMadsen) [inline doc/_static/images/streamlit_chat_input.png:1]: Streamlits font is quite readable compared to ours

File: doc/how_to/streamlit_migration/chat.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+![Streamlit chat_input](../../_static/images/streamlit_chat_input.png)
+
+### Panel Chat Input
+
+Panel does not provide a dedicated *chat input* component because it is built into Panels high-level `ChatInterface`.
+
+Below we will show you how to build a custom `ChatInput` widget.
+
+```python
```
Comment (MarcSkovMadsen) [inline doc/how_to/streamlit_migration/chat.md:70]: Should we add a component like this to Panels chat components @ahuang11 ?

Comment (MarcSkovMadsen) [inline doc/how_to/streamlit_migration/chat.md:70]: This custom `Viewer` component also shows some of the challenges it has.

- The component is not and cannot be used as a widget. I.e. you need to `pn.bind` to and `pn.depends` on `chat_input.param.value` not `chat_input`. There is no `.from_param` method etc.
- It takes a lot of boiler plat code to separate the *viewer* `params` from the *layout* params. And getting `sizing_mode`, `width`, ... properly linked up to the `_layout` is almost impossible.

I.e. you can make custom components using `Viewer`. But I can't see how you can create something that Panel would be willing to include as a widget.

File: doc/how_to/streamlit_migration/chat.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
+# Create Chat Interfaces
```
Comment (MarcSkovMadsen) [inline doc/how_to/streamlit_migration/chat.md:1]: ## Asset for upload

streamlit-status

panel-status

File: doc/how_to/streamlit_migration/chat.md [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+![Streamlit status](https://user-images.githubusercontent.com/42288570/275434382-992f352f-676a-4167-aad0-1fcc2745c130.gif)
+
+### Panel Chat Status
+
+Panel does not provide a dedicated *status* component because it is built into Panels high-level `ChatInterface`. Furthermore Panel provides a lot of other built in *indicators*. Check out the [Indicators Gallery](https://panel.holoviz.org/reference/index.html#indicators).
+
+Below we will show you how to build a custom `Status` indicator.
+
+```python
+import time
```
Comment (MarcSkovMadsen) [inline doc/how_to/streamlit_migration/chat.md:175]: Should we add a `Status` component like this to Panel? We should consider whether we need an extra "ready" `value` option: How do we want to show this status if we have not yet started a run. Should is be disabled, not visible or have some custom look?

**Comment by MarcSkovMadsen** [comment]
I will be on holiday for the next 3 days. I believe what is here is self contained and ready to include if you decide to released the Chat components within the next 3 days.

I plan to add more examples like `CallbackHandler` when I get back from holiday.

File: doc/how_to/streamlit_migration/chat.md (hunk 3 above)
Comment (MarcSkovMadsen) [reply doc/how_to/streamlit_migration/chat.md:183]: Maybe object and objects would be more in line with Panels kaminhylden convention than step and steps?

**Comment by philippjfr** [comment]
Superseded by #5670

**Pull request closed without merging**